src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -1597,22 +1600,25 @@ export class SearchView extends ViewPane {
 			}
 		};
 
-		let visibleMatches = 0;
+		this._visibleMatches = 0;
```

Comment:
I also don't see what this helps

Reply:
> Unless it's possible to start a search before the previous one ends, which is suspicious

Yup, that's what happens. When you type too quickly, it can trigger more than one search at once, which refreshes the tree before the interval clears.

Reply:
> I also don't see what this helps

It's needed for the changes. Since the handler for `_uiRefreshHandle` is now global to the task, we need to also ensure that we're only setting one reference of `visibleMatches`

Reply:
> When you type too quickly, it can trigger more than one search at once

Isn't that a problem on its own though?

Reply:
Where is the code that cancels the search if one is in progress?

Reply:
I haven't looked at this code in forever but it looks like here we try to cancel the current search then wait before starting the next search

https://github.com/microsoft/vscode/blob/ee73b6543eaeded5e7e81a1a70fe75008ea313f9/src/vs/workbench/contrib/search/browser/searchView.ts#L1451-L1454

There might be more than one place where this sort of thing has to happen though

Reply:
I'd imagine that there's weird async stuff happening with `onQueryTriggered` being triggered many times in a row and being unable to cancelSearch in time (?). I can look into it, but maybe in a different PR. I'll create an issue for it.

Reply:
https://github.com/microsoft/vscode/issues/162754